### PR TITLE
School Library - Set Up Associations

### PR DESCRIPTION
**Pull request by faizi2500**
## Following tasks have been completed in this milestone:

- [x] Create a class Classroom with the following
- @label instance variable

- [x] Create a class Book with the following
- @title and @author instance variables

- [x] Create a class Rental with the following
- @date instance variable

- [x] Create the has-many/belongs-to relationship between Classroom and Student
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).

- [x] Create the many-to-many relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-  attr_reader :id, :corrector
+  attr_reader :id, :corrector, :rented
 
   def initialize(age, name, permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @corrector = Corrector.new(@name)
     @age = age
     @permission = permission
+    @rented = []
```
Comment (Mihndim2020) [inline person.rb:14]: - [ ] As it is a  requirement for this project, please a method for adding rentals, `the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association`. Checkout this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) and kindly fix that.

File: books.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Books
+  attr_accessor :book, :author
+  attr_reader :rented
+
+  def initialize(book, author)
+    @book = book
+    @author = author
+    @rented = []
+  end
```
Comment (Mihndim2020) [inline books.rb:9]: - [ ] As it is a  requirement for this project, please a method for adding rentals, `the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.` Checkout this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) and kindly fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Good job passing the linter checks

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments below under the review._

- [ ] Please kindly update your readme file to reflect the changes for this milestone. Please follow this [template](https://github.com/microverseinc/readme-template) and adapt it to your project. 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @faizi2500, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**